core/domain/auth_domain.py

```diff
@@ -31,3 +33,34 @@
 # corresponding Oppia-generated IDs in our APIs.
 AuthIdUserIdPair = (
     collections.namedtuple('AuthIdUserIdPair', ['auth_id', 'user_id']))
+
+
+class AuthClaims(python_utils.OBJECT):
+    """Domain object representing essential Claims (a piece of information about
+    a user, like name, address, phone number, etc.) of an authorized user.
+
+    Attributes:
+        auth_id: str|None. A unique identifier associated to the user. The ID is
+            only unique with respect to the Identity Provider who produced it.
+        email: str|None. The email address associated to the user.
+    """
+
+    def __init__(self, auth_id, email):
+        self.auth_id = auth_id
+        self.email = email
+
+    def __repr__(self):
+        return 'AuthClaims(auth_id=%r, email=%r)' % (self.auth_id, self.email)
+
+    def __hash__(self):
+        return hash((self.auth_id, self.email))
+
+    def __eq__(self, other):
+        return (
+            # https://docs.python.org/2/library/constants.html#NotImplemented
+            NotImplemented if not isinstance(other, AuthClaims) else
+            self.auth_id == other.auth_id and self.email == other.email)
+
+    def __ne__(self, other):
+        # Required in Python 2: https://stackoverflow.com/a/30676267/4859885.
```

Comment:
What happens when we migrate to Python 3 -- will this break?

Reply:
Nope, but we won't need this anymore. Added a TODO to give more context.